File: backend/schemas/campanas.py
from datetime import datetime
from typing import List, Optional

from pydantic import BaseModel


class CampanaLanzarRequest(BaseModel):
    cliente_ids: List[int]
    asunto: str
    cuerpo: str
    remitente: str
    nombre: Optional[str] = None
    delay_segundos: int = 30


class PlantillaResponse(BaseModel):
    id: int
    public_id: str
    nombre_interno: str
    asunto: str
    cuerpo_html: str
    fecha_creacion: datetime | None = None

    class Config:
        from_attributes = True


class CampanaResponse(BaseModel):
    id: int
    public_id: str
    nombre: str
    fecha_creacion: datetime | None = None
    fecha_inicio: datetime | None = None
    fecha_fin: datetime | None = None
    estado: str
    plantilla_id: int | None = None
    remitente: str | None = None
    delay_segundos: int | None = None
    total_destinatarios: int | None = None

    class Config:
        from_attributes = True


class EnvioLogResponse(BaseModel):
    id: int
    campana_id: int | None = None
    cliente_id: int | None = None
    fecha_envio: datetime | None = None
    estado: str
    detalle_error: str | None = None
    intentos: int | None = None

    class Config:
        from_attributes = True

File: frontend/src/types/campanas.ts
export interface CampanaResumen {
	campana_id: number;
	nombre: string;
	estado: string;
	total: number;
	pendientes: number;
	enviados: number;
	errores: number;
	omitidos: number;
	procesados: number;
	finalizado: boolean;
	fecha_creacion: string | null;
	fecha_inicio: string | null;
	fecha_fin: string | null;
	remitente: string | null;
	delay_segundos: number | null;
}

export interface EstadoCampana extends CampanaResumen {
	task_id: string | null;
	mensaje: string;
	log_actividad: string[];
	detener?: boolean;
}

export interface LogCampana {
	id: number;
	cliente_id: number | null;
	email: string | null;
	nombre: string | null;
	estado: string;
	fecha_envio: string | null;
	detalle_error: string | null;
	intentos: number | null;
}

export interface CampanaDetalle {
	campana_id?: number;
	id?: number;
	nombre?: string | null;
	asunto?: string | null;
	cuerpo_html?: string | null;
	remitente?: string | null;
	delay_segundos?: number | null;
	estado?: string | null;
	adjuntos_genericos?: string[] | null;
	adjuntos_guardados?: string[] | null;
	plantilla?: {
		asunto?: string | null;
		cuerpo_html?: string | null;
	} | null;
}

export interface VariableToken {
	label: string;
	token: string;
	ayuda: string;
}

export interface AdjuntoDisponible {
	nombre: string;
	size_bytes: number;
}
